sdk/python/tests/run_test_server.py

```diff
@@ -452,12 +453,16 @@ def run_nginx():
 
     env = os.environ.copy()
     env['PATH'] = env['PATH']+':/sbin:/usr/sbin:/usr/local/sbin'
+    os.mkfifo(nginxconf['ACCESSLOG'], 0700)
```

Comment:
Unfortunately, you can lose both ways, right?  If you lose such that cat never dies (e.g., because nginx fails to start), then reusing an existing FIFO is a bad idea.  But if you lose such that cat does die but the FIFO isn't removed (e.g., the system shuts down between when the tests call `run_nginx` and when they call `stop_nginx`), reporting that as an error is not so helpful.

Normally everything in TEST_TMPDIR we rewrite with impunity.  Since you don't have that option, I think you want to avoid it completely and instead use mktmpfifo.  Or, since that doesn't exist, mkdtemp and put the FIFO in there.  `stop_nginx` can still clean up as you describe, but we'll also recover well from more dramatic failures.